vktrace/vktrace_layer/vktrace_lib_pageguard.cpp

```diff
@@ -388,6 +406,27 @@ LONG WINAPI PageGuardExceptionHandler(PEXCEPTION_POINTERS ExceptionInfo) {
     pageguardExit();
     return resultCode;
 }
+#elif defined(ANDROID)
+void PageGuardExceptionHandler(int sig, siginfo_t* si, void* unused) {
+    if (sig == SIGSEGV) {
+        VkDeviceSize OffsetOfAddr;
+        PBYTE pBlock;
+        VkDeviceSize BlockSize;
+        PBYTE addr = (PBYTE)si->si_addr;
+        LPPageGuardMappedMemory pMappedMem =
+            getPageGuardControlInstance().findMappedMemoryObject(addr, &OffsetOfAddr, &pBlock, &BlockSize);
+        if (pMappedMem) {
+            uint64_t index = pMappedMem->getIndexOfChangedBlockByAddr(addr);
+
+            pMappedMem->setMappedBlockChanged(index, true, BLOCK_FLAG_ARRAY_CHANGED);
+
+            if (mprotect(pMappedMem->getMappedDataPointer() + index * pageguardGetSystemPageSize(),
+                         (SIZE_T)pMappedMem->getMappedBlockSize(index), (PROT_READ | PROT_WRITE)) == -1) {
+                vktrace_LogError("Clear memory protect on page(%d) failed !", index);
+            }
+        }
```

Comment:
If pMappedMem is NULL then this should chain to any previous signal handler.
If g_old_sa is not NULL, then it can be called to deal with the signal.
If g_old_sa is NULL, then returning will only repeat the same instruction that caused the signal.
That will loop endlessly.  It would be better to abort.

Reply:
Thanks, patched.:)